plugins/linux-pulseaudio/pulse-input.c

```diff
@@ -429,10 +423,23 @@ static void pulse_defaults(obs_data_t *settings, bool input)
 {
 	pulse_init();
 
-	pa_server_info_cb_t cb = (input)
-		? pulse_input_device : pulse_output_device;
-	pulse_get_server_info(cb, (void *) settings);
+	const char *device_name;
+	struct pulse_devices *d = bzalloc(sizeof(struct pulse_devices));
+
+	pulse_get_server_info((pa_server_info_cb_t) pulse_default_devices,
+		(void *) d);
+
+	if (input) {
+		device_name = d->default_source_name;
+		blog(LOG_DEBUG, "Default input device: '%s'", device_name);
+	} else {
+		device_name = d->default_sink_name;
+		blog(LOG_DEBUG, "Default output device: '%s'", device_name);
+	}
+
+	obs_data_set_default_string(settings, "device_id", device_name);
 
+	bfree(d);
```

Comment:
You did not free the default_source_name and default_sink_name which are (should be) allocated in the callback. This will create a memory leak.